pac4j-saml/src/main/java/org/pac4j/saml/sso/impl/SAML2DefaultResponseValidator.java

```diff
@@ -523,23 +527,29 @@ protected final boolean isValidBearerSubjectConfirmationData(final SubjectConfir
             return false;
         }
 
-        if (data.getRecipient() == null) {
-            logger.debug("SubjectConfirmationData recipient cannot be null for Bearer confirmation");
-            return false;
-        } else {
-            final Endpoint endpoint = context.getSAMLEndpointContext().getEndpoint();
-            if (endpoint == null) {
-                logger.warn("No endpoint was found in the SAML endpoint context");
-                return false;
-            }
-
-            final String url = endpoint.getLocation();
-            if (!data.getRecipient().equals(url)) {
-                logger.debug("SubjectConfirmationData recipient {} does not match SP assertion consumer URL, found",
-                        data.getRecipient());
+        try {
+            if (data.getRecipient() == null) {
+                logger.debug("SubjectConfirmationData recipient cannot be null for Bearer confirmation");
                 return false;
+            } else {
+                final Endpoint endpoint = context.getSAMLEndpointContext().getEndpoint();
+                if (endpoint == null) {
+                    logger.warn("No endpoint was found in the SAML endpoint context");
+                    return false;
+                }
+    
+                final URI recipientUri = new URI(data.getRecipient());
+                final URI appEndpointUri = new URI(endpoint.getLocation());
+                if (!UriUtils.urisEqualAfterPortNormalization(recipientUri, appEndpointUri)) {
```

Comment:
I looked into org.apache.http.client.utils.URIUtils and it doesn't do that.

It could be done with simple URL.equals() but there are reasons why not to use URL for comparison, please see http://stackoverflow.com/questions/3771081/proper-way-to-check-for-url-equality

Maybe there is a library that does exactly what I wrote but I wasn't able to find it.